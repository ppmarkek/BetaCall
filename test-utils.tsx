import { ReactElement, FC } from 'react';
import { ChakraProvider, defaultSystem } from '@chakra-ui/react';
import { render, RenderOptions } from '@testing-library/react';

const Wrapper: FC<{ children: React.ReactNode }> = ({ children }) => (
  <ChakraProvider value={defaultSystem}>{children}</ChakraProvider>
);

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: Wrapper, ...options });

export * from '@testing-library/react';
export { customRender as render };
