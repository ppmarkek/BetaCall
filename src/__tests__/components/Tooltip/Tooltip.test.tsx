import React from 'react';
import { render } from '../../../../test-utils';
import '@testing-library/jest-dom';
import { Tooltip } from '@/components/ui/tooltip';

describe('Tooltop Component', () => {
  it('render the tooltip with disabled prop', () => {
    const { container } = render(
      <Tooltip content={undefined} disabled>
        Test
      </Tooltip>
    );
    expect(container).toMatchSnapshot();
  });
});
