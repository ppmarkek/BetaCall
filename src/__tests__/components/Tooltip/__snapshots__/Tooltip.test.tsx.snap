// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltop Component render the tooltip with disabled prop 1`] = `
<div>
  Test
</div>
`;
